sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -136,8 +139,22 @@ class AstBuilder extends DataTypeAstBuilder
       label: Option[String],
       allowVarDeclare: Boolean): CompoundBody = {
     val buff = ListBuffer[CompoundPlanStatement]()
+    val handlers = ListBuffer[ErrorHandler]()
```

Comment:
we miss a logic for statement ordering? i.e. if I recall correctly, we should have variable and condition declarations first, followed by handler declarations, and the actual body afterwards?

Reply:
Yes, that check is missing at the moment. It will be added.